dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1199 @@
+                                          self._project_id)
+
+    if not matched_resource:
+      self.logger.error('Resource not found')
+      return
+
+    resource = matched_resource
+
+    # Save resource tree to temp file
+    output_file = tempfile.NamedTemporaryFile(mode='w',
+                                              delete=False,
+                                              encoding='utf-8',
+                                              suffix='.txt')
+    output_path = output_file.name
+
+    self.PublishMessage(f'Saving resource tree to {output_path}')
+    with open(output_path, 'w') as out_file:
+      if resource.parent:
+        out_file.write(str(resource.parent))
+      else:
+        out_file.write(str(resource))
+
+    # Dump the resource tree to CLI
+    if resource.parent:
+      self.PublishMessage(str(resource.parent))
+    else:
+      self.PublishMessage(str(resource))
+
+  def _GetListOfResources(self, project_id: str) -> None:
+    """Acquire a list of resources under a project.
+
+    Args:
+      project_id: Project id to get list of resources from.
+    """
+    # Retrieve list of disks in a project
+    self._resources_dict.update(self._RetrieveListOfDisks(project_id))
+
+    # Retrieve list of disk images in a project
+    self._resources_dict.update(self._RetrieveListOfDiskImages(project_id))
+
+    # Retrieve list of snapshots in a project
+    self._resources_dict.update(self._RetrieveListOfSnapshots(project_id))
+
+    # Retrieve list of instances in a project
+    self._resources_dict.update(self._RetrieveListOfInstances(project_id))
+
+    # Retrieve list of instance templates in a project
+    self._resources_dict.update(
+        self._RetrieveListOfInstanceTemplate(project_id))
+
+    # Retrieve list of machine images in a project
+    self._resources_dict.update(self._RetrieveListOfMachineImages(project_id))
+
+  def _GetResourcesMetaDataFromLogs(self, project_id: str) -> None:
+    """Enrich resources with meta data from GCP Logs.
+
+    Args:
+      project_id: Project id to get list of resources from.
+    """
+    time_ranges: List[Dict[str, datetime]] = []
+
+    all_active_resources = sorted(
+        self._resources_dict.values(),
+        key=lambda resource: resource.creation_timestamp)  # type: ignore
+
+    # Building list of timestamp ranges to query
+    time_range = {}
+    if all_active_resources[0].creation_timestamp:
+      time_range['start_timestamp'] = all_active_resources[
+          0].creation_timestamp - timedelta(hours=1)
+      time_range['end_timestamp'] = all_active_resources[
+          0].creation_timestamp + timedelta(hours=1)
+      time_ranges.append(time_range)
+
+    idx = 1
+    while idx < len(all_active_resources) - 1:
+      time_range = {}
+
+      resource_creation_timestamp = all_active_resources[
+          idx].creation_timestamp
+
+      if not resource_creation_timestamp:
+        return
```

Comment:
do you mean here return or continue?
return will exit the method, continue will jump to the next item in the loop